libcontainer/container_linux_test.go

```diff
@@ -297,19 +297,24 @@ func TestGetContainerState(t *testing.T) {
 		}
 	}
 	for _, ns := range container.config.Namespaces {
-		path := state.NamespacePaths[ns.Type]
+		path, ok := state.NamespacePaths[ns.Type]
+		if !configs.IsNamespaceSupported(ns.Type) {
+			if ok {
+				t.Fatalf("namespace %s is not supported", ns.Type)
+			}
+			continue
+		}
```

Comment:
I would do something like this instead
```go
path := state.NamespacePaths[ns.Type]
if path == "" {
	t.Fatalf("expected non nil namespace path for %s", ns.Type)
}
if !configs.IsNamespaceSupported(ns.Type) {
	t.Fatalf("namespace %s not supported but its path is non-empty: %s", ns.Type, path)
}
```

This is basically the same check as you added, but with less changes to the existing code -- easier to review.